Assets/Tests/InventoryTests.cs

```diff
@@ -0,0 +1,154 @@
+using System.Collections.Generic;
+using Enums;
+using Models;
+using NUnit.Framework;
+
+namespace Tests
+{
+    public class InventoryTests
+    {
+        // A Test behaves as an ordinary method
+        [Test]
+        public void Create_Inventory_ReturnsZeroResources()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            Dictionary<InventoryItems, int> expected = new Dictionary<InventoryItems, int>
+            {
+                {InventoryItems.Wood,0},
+                {InventoryItems.Lenses,0},
+                {InventoryItems.Clay,0},
+                {InventoryItems.Gold,0},
+                {InventoryItems.Steel,0},
+                {InventoryItems.Insulation,0},
+                {InventoryItems.Fertilizer,0},
+                {InventoryItems.Stone,0}
+            };
+            
+            //Then
+            Assert.AreEqual(expected,inventory.GetInventory());
+        }
+    
+        [Test]
+        public void Add_OneResources_ReturnsSameNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,1);
+            
+            //Then
+            Assert.AreEqual(1,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Add_ZeroResources_ReturnsSameNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,0);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Remove_FromEmptyInventory_ReturnsZero()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.RemoveFromInventory(InventoryItems.Wood,1);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Remove_FromFilledInventory_ReturnsLowerNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,2);
+            inventory.RemoveFromInventory(InventoryItems.Wood,2);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void CanBuild_EmptyInventory_ReturnsFalse()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //Then
+            Assert.IsFalse(inventory.CanBuild(InventoryItems.Wood));
```

Comment:
Always use variables in asserts not methods. Makes debugging easier when needed.